astropy/units/format/__init__.py

```diff
@@ -43,6 +50,22 @@
 ]
 
 
+@runtime_checkable
```

Comment:
Move these protocols to a new file `units/format/api.py` ?
IMO they shouldn't be defined in `__init__`.